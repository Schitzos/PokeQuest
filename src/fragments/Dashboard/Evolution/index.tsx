import React from 'react';
import {View, StyleSheet, Animated} from 'react-native';
import {usePokemon} from '@/hooks/usePokemon';
import FastImage from 'react-native-fast-image';
import TextView from '@/components/TextView';
import theme from '@/theme';

export default function Evolution({
  viewEvolveAnimate,
  viewEvolveSpark,
}: {
  viewEvolveSpark: Animated.Value;
  viewEvolveAnimate: Animated.Value;
}) {
  const {pokemon} = usePokemon();

  return (
    <Animated.View style={[styles.base, {opacity: viewEvolveAnimate}]}>
      <View style={styles.evolveContainer}>
        <View style={styles.evolveWrapper}>
          <FastImage
            source={{
              uri: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemon.detail.id}.png`,
              priority: FastImage.priority.high,
            }}
            style={styles.artEvolve}
            defaultSource={require('@assets/images/default_image_loading.png')}
            resizeMode={FastImage.resizeMode.contain}
          />
          <TextView color={theme.colors.white} fz={24}>
            {pokemon.selected.pokemonName}
          </TextView>
        </View>
      </View>
      <Animated.View
        style={[styles.evolveContainer, {opacity: viewEvolveSpark}]}>
        <View style={styles.evolveWrapper}>
          <FastImage
            source={require('@assets/images/evolve-animation.gif')}
            style={styles.artEvolveSpark}
            resizeMode={FastImage.resizeMode.contain}
          />
        </View>
      </Animated.View>
    </Animated.View>
  );
}

const styles = StyleSheet.create({
  base: {
    flex: 1,
    position: 'absolute',
    backgroundColor: 'black',
    width: '100%',
    height: '100%',
  },
  evolveContainer: {
    position: 'absolute',
    flex: 1,
    width: '100%',
    height: '100%',
  },
  evolveWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
  artEvolve: {
    width: 300,
    height: 300,
  },
  artEvolveSpark: {
    width: 500,
    height: 500,
  },
});
