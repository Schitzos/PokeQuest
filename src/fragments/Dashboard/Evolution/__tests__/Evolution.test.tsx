// Evolution.test.tsx

import React from 'react';
import {render} from '@testing-library/react-native';
import Evolution from '../index';
import {Animated} from 'react-native';

jest.mock('@/hooks/usePokemon', () => ({
  usePokemon: () => ({
    pokemon: {
      detail: {id: 25}, // Mocking detail.id for testing purposes
      selected: {pokemonName: 'Pikachu'}, // Mocking selected.pokemonName for testing purposes
    },
  }),
}));

describe('Evolution component', () => {
  test('matches snapshot', () => {
    const viewEvolveAnimate = new Animated.Value(1);
    const viewEvolveSpark = new Animated.Value(1);
    const {toJSON} = render(
      <Evolution
        viewEvolveAnimate={viewEvolveAnimate}
        viewEvolveSpark={viewEvolveSpark}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  // Add more test cases as needed to cover different scenarios and behaviors of the component
});
