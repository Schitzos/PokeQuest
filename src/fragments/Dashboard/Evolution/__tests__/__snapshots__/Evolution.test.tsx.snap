// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evolution component matches snapshot 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "black",
      "flex": 1,
      "height": "100%",
      "opacity": 1,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "height": 300,
              "width": 300,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={
            {
              "testUri": "../../../assets/images/default_image_loading.png",
            }
          }
          resizeMode="contain"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontFamily": "PixeloidMono",
              "fontSize": 24,
              "fontWeight": "normal",
              "textAlign": "left",
              "textTransform": "capitalize",
            },
          ]
        }
      >
        Pikachu
      </Text>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "flex": 1,
        "height": "100%",
        "opacity": 1,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "height": 500,
              "width": 500,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="contain"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
